### DISCUSSION TITLE
Feature: Time Zone Conversion

### DISCUSSION THREAD

**Discussion by stuartpurgavie** [kazad/instacalc#36] [Ideas]
One of the things I often need to do is figure out is what time something is in another time zone or city. For example, I'd like to be able to type into the calculator some of the following statements and get answers:

| What I type | Potential result |
| ---------------- | ------------------------|
| 2100 PST in Mumbai, India | 0930 IST next day (+12.5 hours) |
| 1:00 PM NST at MSK | 6:30 PM MSK same day (+5.5 hours) |
| Noon for Casablanca in Hawaii | One hour past Midnight same day (-11.0 hours) |

I don't know how much the response should match the phrasing of the query or what is reasonable to expect there, but that is the core of the idea. As a stretch goal, you could specify a day of the year like `on January 1` and check to see if some form of daylight savings time applies to either of the time zones in question on that day, but otherwise just do it for the current date.

Thanks for an awesome tool, have a lot of fun with it and definitely get use out of it!

**Comment by kazad** [comment]
Thanks for the suggestion, glad you're enjoying it!

I've long wanted to add date/time/timezone conversions. I'll need to think about the design a bit, and most likely will rely on an external JS date library to bring in a variety of features. Appreciate the suggestion!

**Comment by kazad** [comment]
Quick update, I have a v0.5 for the timezone feature. More details here:

https://github.com/kazad/instacalc/discussions/52 

demo:

https://instacalc.com/56749

I'd like to add more city & country detection, but for now the basic timezone conversions are there.
